src/lib/fauna-command.js

```diff
@@ -87,13 +87,22 @@ class FaunaCommand extends Command {
     }
   }
 
-  mapConnectionError({ err, connectionOptions }) {
+  mapConnectionError({ err, connectionOptions, version }) {
     if (err instanceof errors.Unauthorized) {
       this.error(
         `Could not Connect to ${connectionOptions.url} Unauthorized Secret`
       );
+    } else if (
+      err.description === "UnknownError" &&
+      err?.requestResult?.statusCode === 410 &&
+      version === "4"
+    ) {
+      this.error(
+        `This account is not allowed to query Fauna v4. Please use the v10 endpoint.`
+      );
+    } else {
+      this.error(err);
```

Comment:
This seems error prone, in that it assumes all unknown 410 errors are due to v4 being disabled. I think you should be able to get access to more error details via requestResult rather than relying on heuristics

Reply:
Ashton and I tested locally and found no reason not to just forward the nice message from core: https://github.com/fauna/core/blob/33ae43bb9aeabdea0f596020b0514975020dcf9b/ext/api/src/main/scala/api/ExceptionResponseHelpers.scala#L68

Updated this block to just properly handle `description` and `message`